code/modules/mob/living/basic/space_fauna/paper_wizard/paper_wizard.dm

```diff
@@ -0,0 +1,196 @@
+	grant_abilities()
+	grant_loot()
+	RegisterSignal(src, COMSIG_MOVABLE_PRE_MOVE, PROC_REF(animate_step))
+
+/mob/living/basic/paper_wizard/proc/grant_abilities()
+	summon = new(src)
+	summon.Grant(src)
+	ai_controller.set_blackboard_key(BB_WIZARD_SUMMON_MINIONS, summon)
+	mimic = new(src)
+	mimic.Grant(src)
+	ai_controller.set_blackboard_key(BB_WIZARD_MIMICS, mimic)
+
+/mob/living/basic/paper_wizard/proc/grant_loot()
+	AddElement(/datum/element/death_drops, dropped_loot)
+
+/mob/living/basic/paper_wizard/proc/animate_step()
+	SIGNAL_HANDLER
+	var/paper_effect = new /obj/effect/temp_visual/paper_scatter(get_turf(src))
+	animate(paper_effect, alpha = 0, 1 SECONDS)
+
+/mob/living/basic/paper_wizard/Destroy()
+	QDEL_NULL(summon)
+	QDEL_NULL(mimic)
+	return ..()
+
+/datum/ai_controller/basic_controller/paper_wizard
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic,
+		BB_WRITING_LIST = list(
+			"I can turn the paper into gold and ink into diamonds!",
+			"Your fate is written and sealed!",
+			"You shall suffer the wrath of a thousand paper cuts!",
+		)
+	)
+
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk/less_walking
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/targeted_mob_ability/wizard_mimic,
+		/datum/ai_planning_subtree/use_mob_ability/wizard_summon_minions,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree,
+		/datum/ai_planning_subtree/attack_obstacle_in_path/paper_wizard,
+		/datum/ai_planning_subtree/find_paper_and_write,
+	)
+
+/datum/ai_planning_subtree/attack_obstacle_in_path/paper_wizard
+	target_key = BB_FOUND_PAPER
+	attack_behaviour = /datum/ai_behavior/attack_obstructions/paper_wizard
+
+/datum/ai_behavior/attack_obstructions/paper_wizard
+	action_cooldown = 0.4 SECONDS
+	can_attack_turfs = TRUE
+	can_attack_dense_objects = TRUE
+
+/datum/ai_planning_subtree/targeted_mob_ability/wizard_mimic
+	ability_key = BB_WIZARD_MIMICS
+	finish_planning = FALSE
+
+/datum/ai_planning_subtree/use_mob_ability/wizard_summon_minions
+	ability_key = BB_WIZARD_SUMMON_MINIONS
+	finish_planning = FALSE
+
+/datum/ai_behavior/find_and_set/empty_paper
+	action_cooldown = 10 SECONDS
+
+/datum/ai_behavior/find_and_set/empty_paper/search_tactic(datum/ai_controller/controller, locate_path, search_range)
+	var/list/empty_papers = list()
+
+	for(var/obj/item/paper/target_paper in oview(search_range, controller.pawn))
+		if(target_paper.is_empty())
+			empty_papers += target_paper
+
+	if(empty_papers.len)
+		return pick(empty_papers)
+
+/mob/living/basic/paper_wizard/copy
+	desc = "'Tis a ruse!"
+	health = 1
+	maxHealth = 1
+	alpha = 200
+	faction = list(FACTION_STICKMAN)
+	melee_damage_lower = 1
+	melee_damage_upper = 5
+
+	ai_controller = /datum/ai_controller/basic_controller/wizard_copy
+	///the master that summoned us
+	var/mob/living/basic/paper_wizard/original
```

Comment:
It seems like we don't actually do anything with this reference. We can probably just not set it?